core/krylovObservables.cpp

```diff
@@ -1,50 +1,238 @@
+#include "krylovObservables.h"
 
-#include <iostream>
-#include <complex>
-#include <memory>
+using namespace TE;
 
-#define MKL_Complex16 std::complex<double>
-#define MKL_INT size_t
 
-#include <mkl.h>
-#include <mkl_spblas.h>
+/**
+* Function to return the name of the observable
+* @return Name of the observable
+*/
+std::string krylovBasicObservable::retName()
+{
+	return obs_name;
+}
 
-#include "matrixDataTypes.h"
-#include "krylovObservables.h"
+/**
+* Function to return a pointer to the expectation value array
+* @return Raw pointer to the array which stores the expectation values
+*/
+double* krylovBasicObservable::retexpectationValues()
+{
+    return expectationValues;
+}
 
+/**
+* Function to return the number of samples taken
+* @return Number of samples taken
+*/
+size_t krylovBasicObservable::retnumSamples()
+{
+    return numSamples;
+}
 
 
+/**
+ * Initializes a raw array to requested size for storing the expectation values of the oberservable 
+ * @param size number of entries in the result)
+ */
+void krylovBasicObservable::initializeResultArray(size_t size)
+{	
+	numSamples = size;
+	expectationValues = new double[size];
+}
 
-std::string krylovBasicObservable::retName()
+/**
+* Resets the index of sampling to zero. Note that the original values are not overwritten! 
+* @return The size of the expectation value array, i.e. the number of expectation values that can be recorded.
+*/
+size_t krylovBasicObservable::resetResultArray()
 {
-	return obs_name;
+    sampleIndex = 0;
+    return numSamples;
+}
+
+
+/**
+ * Writes sampled expectation values of a list of observables to file
+ * @param obs_list List of observables with sampled expectation values
+ * @param para List of parameters that are included as attributes in the HDF5 file as well as filename (if requested)
+ * @param name Requested filename
+ */
+void krylovBasicObservable::saveResult(const std::vector<std::unique_ptr<krylovBasicObservable>> &obs_list,  parameter_list &para, const std::string &name)
```

Comment:
Or even better may be to resurrect some of the krylovHelper, i.e. a class in the helper-folder that is given an observable and then outputs it. This has the additional advantage that the core is independent of HDF5. 
But let's see what you think. :)

Reply:
The class parameter is defined in parameter.h which is located in the helper folder. 
I think we agree that krylovObservable should be in core since the the timeEvolver expects one of those as argument. 
Parameter is in helper since although it appears in the argument of saveResult (core) it is completely optional. 
On a side note, krylovObservable is strictly speaking also optional, the timeEvolver will just not output a lot when it's missing (besides the final state).   

We could outsource the saveResult function, which is currently a static class function of krylovObservable, to a separate file in the helper folder. It should not be much work (since it is already a static function). However, I don't think it is an improvement. I also don't have any problems if there is HDF5 code in the core :)

But I am still open to be convinced about it :)